File: scripts/generate_icon.dart
import 'dart:io';
import 'dart:ui' as ui;

void main() async {
  // Create a simple app icon with a checkmark
  final recorder = ui.PictureRecorder();
  final canvas = ui.Canvas(recorder);
  const size = ui.Size(1024, 1024);

  // Background - Primary color #4A6CF7
  final paint = ui.Paint()..color = const ui.Color(0xFF4A6CF7);
  canvas.drawRect(ui.Rect.fromLTWH(0, 0, size.width, size.height), paint);

  // White circle background for icon
  final circlePaint = ui.Paint()..color = const ui.Color(0xFFFFFFFF);

  canvas.drawCircle(
    ui.Offset(size.width / 2, size.height / 2),
    size.width * 0.35,
    circlePaint,
  );

  // Checkmark icon
  final checkPaint = ui.Paint()
    ..color = const ui.Color(0xFF4A6CF7)
    ..strokeWidth = 80
    ..strokeCap = ui.StrokeCap.round
    ..style = ui.PaintingStyle.stroke;

  final path = ui.Path();
  path.moveTo(size.width * 0.3, size.height * 0.5);
  path.lineTo(size.width * 0.45, size.height * 0.65);
  path.lineTo(size.width * 0.7, size.height * 0.35);
  canvas.drawPath(path, checkPaint);

  final picture = recorder.endRecording();
  final image = await picture.toImage(1024, 1024);
  final byteData = await image.toByteData(format: ui.ImageByteFormat.png);

  if (byteData != null) {
    final buffer = byteData.buffer.asUint8List();
    final file = File('assets/images/app_icon.png');
    await file.writeAsBytes(buffer);
    print('Icon generated: ${file.path}');
  }
}

File: scripts/generate_icon.py
from PIL import Image, ImageDraw

# Create 1024x1024 icon
size = 1024
img = Image.new('RGBA', (size, size), (74, 108, 247, 255))  # #4A6CF7
draw = ImageDraw.Draw(img)

# White circle background
circle_center = (size // 2, size // 2)
circle_radius = int(size * 0.35)
draw.ellipse(
    [circle_center[0] - circle_radius, circle_center[1] - circle_radius,
     circle_center[0] + circle_radius, circle_center[1] + circle_radius],
    fill=(255, 255, 255, 255)
)

# Draw checkmark
check_color = (74, 108, 247, 255)  # #4A6CF7
line_width = 80

# Checkmark points
p1 = (int(size * 0.3), int(size * 0.5))
p2 = (int(size * 0.45), int(size * 0.65))
p3 = (int(size * 0.7), int(size * 0.35))

# Draw lines
draw.line([p1, p2], fill=check_color, width=line_width)
draw.line([p2, p3], fill=check_color, width=line_width)

# Save
img.save('assets/images/app_icon.png')
print(f"Icon saved to assets/images/app_icon.png ({size}x{size})")
